casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -178,7 +178,7 @@ final case class BlockReceiverState[MId: Show] private (
 object BlockReceiver {
   def apply[F[_]: Concurrent: BlockStore: BlockDagStorage: BlockRetriever: Log](
       state: Ref[F, BlockReceiverState[BlockHash]],
-      incomingBlocksStream: Stream[F, BlockMessage],
+      incomingBlocksQueue: Queue[F, BlockMessage],
```

Comment:
The general rule about queues and streams that we want to follow is to not pass queues as input args. Because if `dequeue` is called 2 times on a queue - this can produce unpredictable effects. So the scope of using a queue should be very small, it should be immediately dequeued and turned into a stream. If you have to be able to enqueue something - better pass the callback function `_ => F[Unit]`.